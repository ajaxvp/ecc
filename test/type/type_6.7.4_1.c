/* ISO: 6.7.4; function specifier constraints */

static int z;

inline int g(void)
{
    // ISO: 6.7.4 (3) - valid, const-qualified
    const static int y;
    // ISO: 6.7.4 (3) - invalid, static storage duration but modifiable
    static int x;

    // ISO: 6.7.4 (3) - invalid, internal linkage identifier referenced
    z;
}

// ISO: 6.7.4 (4) - invalid, inline may not be used in declaration of main
inline int main(void);
