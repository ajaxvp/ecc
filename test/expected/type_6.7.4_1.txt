ecc: error: [10:16] an inline function may not declare a non-const identifier with static storage duration
ecc: error: [13:5] an inline function may not contain a reference to an identifier declared with internal linkage
ecc: error: [17:12] 'main' should not have the 'inline' function specifier
